server/src/Utopia/Web/Packager/NPM.hs

```diff
@@ -176,8 +199,77 @@ getProjectDependenciesFromPackageJSON decodedProject = either (const []) identit
   contentsTreeRoot <- first toS $ projectContentTreeFromDecodedProject decodedProject
   packageJsonFile <- maybe (Left "No package.json found.") pure $ getProjectContentsTreeFile contentsTreeRoot ["package.json"]
   packageJsonCode <- maybe (Left "package.json not a text file.") pure $ firstOf projectFileToCodeLens packageJsonFile
-  MinimalPackageJSON{..} <- eitherDecode' $ BL.fromStrict $ encodeUtf8 packageJsonCode
-  let fullDependencies = fromMaybe mempty (dependencies <> devDependencies)
-  pure $ Map.foldMapWithKey (\key value -> [ProjectDependency key value]) fullDependencies
+  packageJSON <- eitherDecode' $ BL.fromStrict $ encodeUtf8 packageJsonCode
+  pure $ packageJSONToDependencies packageJSON
+
+getPackageJSONContent :: Text -> IO MinimalPackageJSON
+getPackageJSONContent packageJSONURL = do
+  response <- WR.get $ toS packageJSONURL
+  either fail pure $ eitherDecode $ view WR.responseBody response
+
+packageJSONToDependencies :: MinimalPackageJSON -> [ProjectDependency]
+packageJSONToDependencies MinimalPackageJSON{..} =
+  let dependenciesMap = fromMaybe mempty (dependencies <> devDependencies)
+  in  Map.foldMapWithKey (\key value -> [ProjectDependency key value]) dependenciesMap
+
+cachePackagerContent :: (MonadResource m, MonadMask m) => PackageVersionLocksRef -> Text -> ConduitBytes m -> IO (ConduitBytes m, UTCTime)
```

Comment:
there's a non-zero chance that under load this will really stress out the I/O of the actual server storage, but I guess it's a problem for tomorrow

Reply:
This is the already existing function that we haven't seen an issue with so far, just moved to a different file.